lib/tom_queue/deferred_work_manager.rb

```diff
@@ -20,130 +24,36 @@ class DeferredWorkManager
 
     include LoggingHelper
 
-    # Public: Scoped singleton accessor
-    #
-    # This returns the shared work manager instance, creating it if necessary.
-    #
-    # Returns a DeferredWorkManager instance
-    @@singletons_mutex = Mutex.new
-    @@singletons = {}
-    def self.instance(prefix = nil)
-      prefix ||= TomQueue.default_prefix
-      prefix || raise(ArgumentError, 'prefix is required')
-
-      @@singletons_mutex.synchronize { @@singletons[prefix] ||= self.new(prefix) }
-    end
-
-
-    # Public: Return a hash of all prefixed singletons, keyed on their prefix
-    #
-    # This method really is just a convenience method for testing.
-    #
-    # NOTE: The returned hash is both a dupe and frozen, so should be safe to 
-    # iterate and mutate instances.
-    # 
-    # Returns: { "prefix" => DeferredWorkManager(prefix),  ... }
-    def self.instances
-      @@singletons.dup.freeze
-    end
-
-    # Public: Shutdown all managers and wipe the singleton objects.
-    # This method is really just a hook for testing convenience.
-    #
-    # Returns nil
-    def self.reset!
-      @@singletons_mutex.synchronize do
-        @@singletons.each_pair do |k,v|
-          v.ensure_stopped
-        end
-        @@singletons = {}
-      end
-      nil
-    end
```

Comment:
I removed everything related to singletons. The idea is that it's just one `DeferredWorkManager` instance per process